spec/datadog/core/configuration/options_spec.rb

```diff
@@ -114,6 +114,23 @@
           before { options_class.send(:option, name) }
 
           it { expect { set_option }.to change { options_object.send(name) }.from(nil).to(value) }
+
+          it 'defaults to PROGRAMMATIC precedence' do
+            set_option
+
+            expect(options_object.options[name].send(:precedence_set))
+              .to eq(Datadog::Core::Configuration::Option::Precedence::PROGRAMMATIC)
+          end
+
+          context 'with precedence' do
+            subject(:set_option) { options_object.set_option(name, value, precedence: precedence) }
+            let(:precedence) { 123 }
+
+            it 'sets the option precedence' do
+              set_option
+              expect(options_object.options[name].send(:precedence_set)).to eq(precedence)
+            end
+          end
```

Comment:
I wonder if we should enforce that precedence is one of the defined ones? Just to make sure nobody thinks to provide their own custom level and wreck havoc.

Reply:
I agree with that. We should probably validate that the precedence is part of the ones defined by us 😄

Reply:
I changed the test to use a real precedence value, as to not lead people into thinking it can be anything.